packages/react-admin/src/mui/list/Filter.js

```diff
@@ -23,20 +24,21 @@ export class Filter extends Component {
         this.filters = nextProps.filterValues;
     }
 
-    componentWillUnmount() {
-        if (this.props.setFilters) {
-            this.setFilters.cancel();
-        }
-    }
-
-    setFilters = debounce(filters => {
-        if (!isEqual(filters, this.filters)) {
+    handleFilterFormSubmit = (values, dispatch, props) => {
```

Comment:
no, the `FilterButton` should pass a `setFilters` function to the form. It's the form's responsibility to call this function both on change and on submit.